sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLHandler.scala

```diff
@@ -125,6 +127,15 @@ private[connect] object MLHandler extends Logging {
         val dataset = MLUtils.parseRelationProto(fitCmd.getDataset, sessionHolder)
         val estimator =
           MLUtils.getEstimator(sessionHolder, estimatorProto, Some(fitCmd.getParams))
+
+        // pre-training model size check
+        val maxSize = conf.getConf(Connect.CONNECT_SESSION_ML_CACHE_SINGLE_ITEM_SIZE)
+        if (maxSize > 0) {
+          val estimatedSize = estimator.estimateModelSize(dataset)
+          if (estimatedSize > maxSize) {
+            throw MlItemSizeExceededException("fit", estimator.uid, estimatedSize, maxSize)
```

Comment:
similar comment as before.

Reply:
I think we should not stop the session in this case

Reply:
No we should not stop the session. The client can technically recover by increasing the limits.